### PR TITLE
20-10207 - Build front-end - Part 4 

### PR DESCRIPTION
**Pull request by tlei123**
## Summary

- Build front-end (Part 4) of Priority Processing (20-10207):
  - Evidence chapter (aka Upload-circumstances)
- Team: Veteran Facing Forms
- Flipper not implemented

## Related issue(s)

- department-of-veterans-affairs/va.gov-team-forms#940
- department-of-veterans-affairs/va.gov-team-forms#909

## Testing done

- Local browser-, unit-, & e2e-tests all successful

## What areas of the site does it impact?

This form ONLY.

### CODE REVIEW

File: src/applications/simple-forms/20-10207/config/constants.js [hunk 1]
```diff
@@ -283,13 +283,205 @@ export const FINANCIAL_HARDSHIP_DESCRIPTION = Object.freeze(
...
-      data-testid="hardshipAdditionalInfo"
+      data-testid="financialHardshipAdditionalInfo"
     >
       <ul>
         <li>Copy of an eviction notice or statement of foreclosure</li>
         <li>Copy of notices of past-due utility bills</li>
         <li>Copy of collection notices from creditors</li>
       </ul>
     </va-additional-info>
+    <p />
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/20-10207/config/constants.js:294]: same, looks like you could do some refactoring here if you like

File: src/applications/simple-forms/20-10207/helpers.js [hunk 2]
```diff
@@ -122,3 +122,27 @@ export function dateOfDeathValidation(errors, fields) {
     );
   }
 }
+
+export function powConfinementDateRangeValidation(errors, fields) {
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/20-10207/helpers.js:126]: This was another one I was thinking about patternizing, as date ranges are pretty common. like maybe just provide the validation as a pattern where you can specify the two fields.

File: src/applications/simple-forms/20-10207/pages/evidenceALS.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import { createPayload, parseResponse } from '../helpers';
+
+const uiTitle =
+  'Upload evidence for diagnosis of ALS (amyotrophic lateral sclerosis)';
+const uiDescription =
+  'If you have documents you would like to submit as evidence of ALS, upload them here.';
+
+/** @type {PageSchema} */
+export default {
+  uiSchema: {
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/20-10207/pages/evidenceALS.js:17]: not sure if it makes sense to patternize this or not,  for example abstracting all the stuff that will usually stay the same. but this does seem like a lot of duplicate code. even local patterns could be fine.

**Review by rhasselle-oddball — COMMENTED** [review]
Looks good, I think it would be worth considering if we want to refactor all this into local patterns/components, because there's a lot of duplicate code, and those could even eventually become forms-system patterns if we like the refactors

**Review by rhasselle-oddball — APPROVED** [review]

File: src/applications/simple-forms/20-10207/components/AlsViewField.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { displayFileSize } from 'platform/utilities/ui/index';
+
+import { ALS_DESCRIPTION } from '../config/constants';
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/simple-forms/20-10207/components/AlsViewField.jsx:6]: Yeah, that'd be useful.  I'll ping ya on Slack to discuss.  There's a new USWDS File input that we haven't used at all, so I'm not sure how you'd like to approach it.

File: src/applications/simple-forms/20-10207/helpers.js (hunk 2 above)
Comment (tlei123) [reply src/applications/simple-forms/20-10207/helpers.js:126]: Yeah, definitely would be useful to have a range-pattern with range-validation.

File: src/applications/simple-forms/20-10207/pages/evidenceALS.js (hunk 3 above)
Comment (tlei123) [reply src/applications/simple-forms/20-10207/pages/evidenceALS.js:17]: Good point.  I'll try DRYing in part 5.  Thanks!

File: src/applications/simple-forms/20-10207/config/constants.js (hunk 1 above)
Comment (tlei123) [reply src/applications/simple-forms/20-10207/config/constants.js:294]: Agreed.  I'll try refactoring for DRYer in Part 5.  Thanks!

**Pull request merged**